pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -226,6 +226,22 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 	causes = append(causes, validateVSOCK(field, spec, config)...)
 	causes = append(causes, validatePersistentReservation(field, spec, config)...)
 	causes = append(causes, validatePersistentState(field, spec, config)...)
+	causes = append(causes, validateDownwardMetrics(field, spec, config)...)
+
+	return causes
+}
+
+func validateDownwardMetrics(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec, config *virtconfig.ClusterConfig) []metav1.StatusCause {
+	var causes []metav1.StatusCause
```

Comment:
Returning a slice is used as poor man's optional type.

We can't just return "empty" `SatusCause`, otherwise even for a valid VMI the `len(cause)` will be `> 0` and the admitter will fail.

We can return a pointer to it and check for `nil` (or a `bool`), or extract the first half of the `if` so we can call it:
```
if  downwardmetrics.HasDevice(spec) {     // spec.Domain.Devices.DownwardMetrics != nil
     causes = append(causes, validateDownwardMetrics(field, spec, config))
}
```
but I don't really like these options. 

Alternatively, we can change the function to accept a cause slice and return it, so instead of been called:
```
causes = append(causes, validateDownwardMetrics(field, spec, config)...)
```
it should be called
```
causes = validateDownwardMetrics(field, spec, config, causes)
```

I'm more inclined to keep the slice, wdyt?